### DISCUSSION TITLE
Set coordinates for OpenWeatherMap data manually

### DISCUSSION THREAD

**Discussion by maehmann** [jeroenterheerdt/HAsmartirrigation#346] [General]
Hey,

I'm trying to set up your great looking integration for my situation. This is where it get's a bit tricky. I have two locations separated by 70km that run on one HA installation being linked through a VPN. So my Raspi sits in location A but I want to setup the irrigation system for location B which is not configure as the main location in HA obviously.

On location B I do not have a precipitation sensor. Only temp, pressure, humidity, due point. If I understand correctly then all info missing from the sensors will be taken from OpenWeatherMap depending on the set coordinates in HA.

Is this correct? And if so, is there a way to set a new location manually somehow (by coordinates or by a defined zone in HA)?

Thank you for your hard work and in advance for your reply!

Christoph

**Comment by jeroenterheerdt** [comment]
It's not. It gets it automatically from the HA config it's on. So unless you want to edit the code the only option is to change the location in HA.

**Comment by maehmann** [comment]
Any chance that you could add such a functionality? I'm not a programmer unfortunately. Otherwise I would be happy to integrate it myself.

I can't simply change the location, either, as the one where I need the irrigation system is not my main location. 

Thanks for considering.

**Reply by jeroenterheerdt** [reply]
I could. It would then be an override of automatic location?

**Reply by maehmann** [reply]
Thanks for thinking about implementing. That is really appreciated.

Yes, that sounds like a good plan. Entering coordinates directly would be fine or using another zone defined in Home Assistant.

**Comment by maehmann** [comment]
Hello,

Are you seriously considering to add the manual override of the coordinates? I don't want to be pushy and I fully understand you are doing this in your spare time, which is much appreciated. 

I just want to know if I should wait for it or start looking for alternative ways to achieve my goals.

Thanks again for considering and in advance for your answer.

**Comment by jeroenterheerdt** [comment]
Yes, I am, but I don't have time to implement it right away. You can edit the code easily though if you want to get it now. Just find the place it loads the long and lat from HA and put in your own numbers.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by maehmann** [reply]
Thanks. That is great. 

I would change the code myself but I have never written a single line of code. I know where to find the files in HA and can edit them but no I have no idea what I would need to change. 

So, I would appreciate if you would point me to the right file and line number. I also don't know the correct format of the coordinates.

I had a look myself. OWNClient.py and the lines below could be the ones:
56        self.longitude = longitude
57        self.latitude = latitude

Any help is welcome. Thanks!

**Comment by jeroenterheerdt** [comment]
Those are the ones. Also line 58 for altitude. Just replace what is after the = with your data, with a dot as decimal separator. So 123.32

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by maehmann** [reply]
Thanks!

**Reply by motogon** [reply]
Is it possible to edit these values in the UI?
Thanks!

**Reply by jeroenterheerdt** [reply]
No

**Comment by jeroenterheerdt** [comment]
Added issue #351 to track this work.
